rclpy/src/rclpy/wait_set.hpp

```diff
@@ -0,0 +1,115 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLPY__WAIT_SET_HPP_
+#define RCLPY__WAIT_SET_HPP_
+
+#include <pybind11/pybind11.h>
+
+#include <string>
+
+namespace py = pybind11;
+
+namespace rclpy
+{
+/// Return a Capsule pointing to a zero initialized rcl_wait_set_t structure
+py::capsule
+get_zero_initialized_wait_set();
+
+/// Initialize a wait set
+/**
+ * Raises RCLError if the wait set could not be initialized
+ *
+ * \param[in] pywait_set Capsule pointing to the wait set structure
+ * \param[in] node_name string name of the node to be created
+ * \param[in] number_of_subscriptions int
+ * \param[in] number_of_guard_conditions int
+ * \param[in] number_of_timers int
+ * \param[in] number_of_clients int
+ * \param[in] number_of_services int
```

Comment:
Can you add docs for `pycontext`?

Reply:
Documented pycontext in dbc0c2335ef4151d9e346e5915428ee0539a5942